app/gui/src/project-view/providers/interactionHandler.ts

```diff
@@ -44,37 +70,70 @@ export class InteractionHandler {
     if (this.isActive(interaction)) this.currentInteraction.value = undefined
   }
 
-  /** End the current interaction, if it is the specified instance. */
+  /**
+   * End the interaction, if it is currently active.
+   * Any children interactions of the given interaction will be ended as well.
+   */
   end(interaction: Interaction) {
-    if (this.isActive(interaction)) {
-      this.currentInteraction.value = undefined
-      interaction.end?.()
+    search(this.currentInteraction.value, interaction, (found, children) => {
+      this.currentInteraction.value = found.parentInteraction
+      found.end?.()
+      for (const interaction of children) {
+        interaction.end?.()
+      }
+    })
+  }
+
+  /** End all interactions, in the order from child to parent. */
+  endAll() {
+    let current = this.currentInteraction.value
+    while (current != null) {
+      current.end?.()
+      current = current.parentInteraction
     }
+    this.currentInteraction.value = undefined
   }
 
-  /** Cancel the current interaction, if it is the specified instance. */
+  /**
+   * Cancel the interaction, if it is currently active.
```

Comment:
The active is ambiguous, because `isActive` does not consider _parents_ active, but here we do.

Reply:
Rephrased